integrationtest/backend-tck/src/main/java/org/hibernate/search/v6poc/integrationtest/backend/tck/search/predicate/MatchSearchPredicateIT.java

```diff
@@ -0,0 +1,299 @@
+
+		query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().bool()
+						.should().match().onField( "string" ).boostedTo( 42 ).matching( MATCHING_STRING )
+						.should().match().onField( "string" ).matching( NON_MATCHING_STRING )
+				.end()
+				.sort().byScore().end()
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsExactOrder( indexName, MATCHING_ID, NON_MATCHING_ID );
+	}
+
+	@Test
+	public void multi_fields() {
+		IndexSearchTarget searchTarget = indexManager.createSearchTarget().build();
+
+		// onField(...).orField(...)
+
+		SearchQuery<DocumentReference> query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().match().onField( "string" ).orField( "one_more_string" ).matching( MATCHING_STRING )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsAnyOrder( indexName, MATCHING_ID );
+
+		query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().match().onField( "string" ).orField( "one_more_string" ).matching( MATCHING_ONE_MORE_STRING )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsAnyOrder( indexName, MATCHING_ID );
+
+		// onField().orFields(...)
+
+		query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().match().onField( "string" ).orFields( "one_more_string", "even_more_string" ).matching( MATCHING_STRING )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsAnyOrder( indexName, MATCHING_ID );
+
+		query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().match().onField( "string" ).orFields( "one_more_string", "even_more_string" ).matching( MATCHING_ONE_MORE_STRING )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsAnyOrder( indexName, MATCHING_ID );
+
+		query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().match().onField( "string" ).orFields( "one_more_string", "even_more_string" ).matching( MATCHING_EVEN_MORE_STRING )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsAnyOrder( indexName, MATCHING_ID );
+
+		// onFields(...)
+
+		query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().match().onFields( "string", "one_more_string" ).matching( MATCHING_STRING )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsAnyOrder( indexName, MATCHING_ID );
+
+		query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().match().onFields( "string", "one_more_string" ).matching( MATCHING_ONE_MORE_STRING )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsAnyOrder( indexName, MATCHING_ID );
+	}
+
```

Comment:
You should probably also add tests where you try to search on fields that do not exist.